### DISCUSSION TITLE
Marathon Type & functionality

### DISCUSSION THREAD

**Discussion by jcookson-infoblox** [code4recovery/12-step-meeting-list#978] [Ideas]
### Contact Details

jcookson@infoblox.com

### Requested Feature/Enhancment

This is to serve as both an enhancement and feature request.

The enhancement is the addition of a new type of meeting, called "Recovery Marathon".  Keeping the type to be non-fellowship specific will allow it to cross the borders of AA, NA, CA, GA & others.  Because of its type, then the individual fellowship will be free to utilize their form of "-athon" for the name of the meeting.  Example: "Thanksgiving Alc-athon".

The feature request is for what happens next once the type is selected.  As the TSML plugin & the meeting guide app both have access to the internal date / time modules of their respective platforms, the feature can then bring forward a calendar picker to select a start date & time and another to select an end date & time.  Once these selections are made, on the back end, the application will evaluate the selections and create the appropriate number of meetings, spanning the time(s) correctly.

Example: selecting 5 pm thanksgiving eve and 11:59 pm Friday after thanksgiving.  This will generate 3 meetings, one for Wednesday from 5 pm to midnight, one for Thursday from midnight to midnight, and one for Friday from midnight to midnight.

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

**Comment by tim-rohrer** [comment]
An interesting idea, but since I do not believe every area implements alc-athons the same way, I don't think it makes sense for the base TSML. And TSML tries to focus more on recurring meetings on a weekly basis. So meetings that occur once or twice a year don't really fit in this way. I believe marathon meetings are better advertised using announcements and websites.

If you still want a marathon type, my suggestion is to create it locally and then use the notes to explain the pattern.

**Reply by jcookson-infoblox** [reply]
Ok, thanks.  Was an idea that i thought might be implemented as a special type.  Been around for a while and most alc-a-thons that I've known usually go for around a day to a day and a half ... which was the case here.
